operations.go

```diff
@@ -0,0 +1,131 @@
+package tezos
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+// OperationElem must be implemented by all operation elements
+type OperationElem interface {
+	OperationElemKind() string
+}
+
+// GenericOperationElem is a most generic element type
+type GenericOperationElem struct {
+	Kind string `json:"kind"`
+}
+
+// OperationElemKind implements OperationElem
+func (e *GenericOperationElem) OperationElemKind() string {
+	return e.Kind
+}
+
+// OperationElements is a slice of OperationElem with custom JSON unmarshaller
+type OperationElements []OperationElem
+
+// UnmarshalJSON implements json.Unmarshaler
+func (e *OperationElements) UnmarshalJSON(data []byte) error {
+	var raw []json.RawMessage
+
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
+	}
+
+	*e = make(OperationElements, len(raw))
+
+opLoop:
+	for i, r := range raw {
+		var tmp GenericOperationElem
+		if err := json.Unmarshal(r, &tmp); err != nil {
+			return err
+		}
+
+		switch tmp.Kind {
+		case "endorsement":
+			(*e)[i] = &EndorsementOperationElem{}
+
+		default:
+			(*e)[i] = &tmp
+			continue opLoop
+
+			// TODO: add other item kinds
+		}
+
+		if err := json.Unmarshal(r, (*e)[i]); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// EndorsementOperationElem represents an endorsement operation
+type EndorsementOperationElem struct {
+	GenericOperationElem
+	Level    int                           `json:"level"`
+	Metadata *EndorsementOperationMetadata `json:"metadata"`
+}
+
+// EndorsementOperationMetadata represents an endorsement operation metadata
+type EndorsementOperationMetadata struct {
+	BalanceUpdates []*BalanceUpdate `json:"balance_updates"`
+	Delegate       string           `json:"delegate"`
+	Slots          []uint           `json:"slots"`
+}
+
+// BalanceUpdate represents balance update operation
+type BalanceUpdate struct {
+	Kind     string        `json:"kind"`
+	Category string        `json:"category"`
+	Delegate string        `json:"delegate"`
+	Level    int           `json:"level"`
+	Change   BalanceChange `json:"change"`
+	Contract string        `json:"contract"`
+}
+
+// BalanceChange is a string encoded int64
+type BalanceChange int64
```

Comment:
Wouldn't this have to be a bignum as well?

Reply:
According to the Tezos source code it wouldn't. It's an int64 internally